types/src/shared/cache.ts

```diff
@@ -1,30 +1,55 @@
 import { redisClient } from "../shared/redis_client";
 
+// JSON-serializable primitive types.
+type JsonPrimitive = string | number | boolean | null;
+
+// Recursive type to check if a type is JSON-serializable.
+type RecursiveJsonSerializable<T> = T extends JsonPrimitive
+  ? T
+  : T extends Array<infer U>
+  ? RecursiveJsonSerializable<U>[]
+  : T extends object
+  ? { [K in keyof T]: RecursiveJsonSerializable<T[K]> }
+  : never;
+
+// Helper type to check if a type is 'never'.
+type IsNever<T> = [T] extends [never] ? true : false;
+
+/**
+ * Ensures that a type is strictly JSON-serializable.
+ * If T is not JSON-serializable, this type resolves to 'never'.
+ */
+export type JsonSerializable<T> = IsNever<
```

Comment:
To me the `IsNever` is making this more complex

Reply:
why not just `[...] extends never` ? Actually not even, because if T doesn't extend never then it's T and if it does then it's never, so it's always just `T` isn't it ?

Reply:
This is required to enforce that `RecursiveJsonSerializable<T>` is exactly the same as the original type `T`.